catkin_home/src/main_engine/src/Action.py

```diff
@@ -0,0 +1,71 @@
+import rospy
+import actionlib
+import time
+#from object_detection.msg import RecognizeObjectAction, RecognizeObjectGoal
+# TODO: Import all the messages related to actions I guess ?
+'''
+All Goals, and Actions must be imported here from he corresponding packages
+'''
+
+
+class Action(object):
+    def __init__(self, id, priority, action_client_binded, category, args):
+        self.id = id
+        self.priority = int(priority)
+        self.args = args
+        self.category = int(category)
+        #Possible States Are: PENDING, ACTIVE, RECALLED, REJECTED, PREEMPTED, ABORTED, SUCCEEDED, LOST.
+        self.feedback = ["PENDING", "ACTIVE", "RECALLED", "REJECTED", "PREEMPTED", "ABORTED", "SUCCEEDED", "LOST"]
+        self.feedback_step = 0
+        self.action_client_name = action_client_binded 
+        self.action_client  = None
+        self.time_requested = int(time.time())
```

Comment:
Do you think the seconds passed since execution would be useful? I dont know how are you using this but I think it could be better to use utcnow(), time.ctime() or something like that

Reply:
It is just to identify different actions of the same id, for example:
bring_something-12784512
bring-something-46531626